Information/CenterOfMass.FCMacro

```diff
@@ -1017,80 +1059,121 @@ class CenterofmassWidget(QtGui.QWidget):
-            with open(fileName, 'r', encoding='utf-8') as csvfile:
-                app.Console.PrintMessage('Reading from %s\n' % fileName)
-                reader = csv.DictReader(csvfile, delimiter='\t')
-                matLoaded = 0
-                # Syntax: [ EXP for x in seq if COND ]
-                nameOfDensityCol = next((s for s in reader.fieldnames if 'density' in s.lower()), None)
-                try:
-                    # Extract unit between brackets
-                    unitForD = nameOfDensityCol.split('(')[1].split(')')[0]
-                except:
-                    unitForD = self.unitForD
-                for row in reader:
-                    if row['Label'] in selList:
-                        sol = selList.index(row['Label'])
-                        try:
-                            self.solids[sol].combo.setCurrentText(row['Material'])
-                        except:
-                            app.Console.PrintWarning('Unknown material or unable to find a '
-                                + '"Material" column in the file --> Material set to "custom"')
-                            self.solids[sol].combo.setCurrentText('custom')
-                        if self.solids[sol].combo.currentText() == 'custom':
-                            try:
-                                qty = row[nameOfDensityCol] + ' ' + unitForD
-                                self.solids[sol].spinDens.setValue(
-                                    Units.Quantity(qty).getValueAs(self.unitForD))
-                            except:
-                                self.solids[sol].combo.setCurrentText('default')
-                                app.Console.PrintWarning('Unable to find a "Density" column '
-                                    + 'in the file --> Material set to "default"')
-                        matLoaded += 1
-                app.Console.PrintMessage(str(matLoaded) + ' materials loaded')
+        """Load a previously exported or an external bill of materials (BOM)"""
+        encoding, delimiter = 'utf-8', '\t'    # default
+        fileFilter = ('CSV (Tab-separated) (*.csv)','Text (Tab-separated) (*.txt)',
+                      'CSV (Excel, semicolons) (*.csv)')
+        fileName, selectedFilter = QtGui.QFileDialog.getOpenFileName(self,
+            'Open file', os.path.expanduser('~'), ';;'.join(fileFilter))
+        if fileName == '':
+            return
+        if selectedFilter == fileFilter[-1]:
+            import locale
+            encoding = locale.getpreferredencoding()
+            delimiter = ';'
+        with open(fileName, 'r', encoding=encoding) as csvfile:
+            app.Console.PrintMessage('Reading from %s\n' % fileName)
+            reader = csv.DictReader(csvfile, delimiter=delimiter, skipinitialspace=True)
+            col = {}    # first columns for label(s), density/ies, mass(es), material(s)
+            for name in ['label', 'densit', 'mass', 'material']:
```

Comment:
What is the advantage of using `densit` over the more clear `density`?

Reply:
These are the search strings covering singular, plural and compound words